docs/mp/vault/01_Vault.adoc

```diff
@@ -0,0 +1,506 @@
+    <artifactId>helidon-integrations-vault-auths-token</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.vault.auths</groupId>
+    <artifactId>helidon-integrations-vault-auths-approle</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.vault.auths</groupId>
+    <artifactId>helidon-integrations-vault-auths-k8s</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.vault.secrets</groupId>
+    <artifactId>helidon-integrations-vault-secrets-kv1</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.vault.secrets</groupId>
+    <artifactId>helidon-integrations-vault-secrets-kv2</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.vault.secrets</groupId>
+    <artifactId>helidon-integrations-vault-secrets-cubbyhole</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.vault.secrets</groupId>
+    <artifactId>helidon-integrations-vault-secrets-transit</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.vault.secrets</groupId>
+    <artifactId>helidon-integrations-vault-secrets-database</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.vault.sys</groupId>
+    <artifactId>helidon-integrations-vault-sys</artifactId>
+</dependency>
+----
+
+To use Vault integration in MP, the following module must be added, as it enables injection of all features:
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.integrations.vault</groupId>
+    <artifactId>helidon-integrations-vault-cdi</artifactId>
+</dependency>
+----
+Configuration to connect to Vault (using token and AppRole authentication):
+[source,properties]
+----
+vault.default.address=http://localhost:8200
+vault.default.token=my-token
+vault.approle.address=http://localhost:8200
+vault.approle.role-id=app-role-id
+vault.approle.secret-id=app-role-secret-id
+----
+The following classes can be injected into any CDI bean (if appropriate module is on the classpath):
+
+* Kv2Secrets - Key/Value Version 2 Secrets (versioned secrets, default)
+* Kv1Secrets - Key/Value Version 1 Secrets (unversioned secrets, legacy)
+* CubbyholeSecrets - Cubbyhole secrets (token bound secrets)
+* DbSecrets - Database secrets (for generating temporary DB credentials)
+* PkiSecrets - PKI secrets (for generating keys and X.509 certificates)
+* TransitSecrets - Transit operations (encryption, signatures, HMAC)
+* AppRoleAuth - AppRole authentication method (management operations)
+* K8sAuth - Kubernetes authentication method (management operations)
+* TokenAuth - Token authentication method (management operations)
+* Sys - System operations (management of Vault - enabling/disabling secret engines and authentication methods)
+* *Rx - reactive counterparts to the classes defined above, usually not recommended in CDI, as it is a blocking environment
+
+
+== Usage
```

Comment:
Here we should have a single example of how to inject/use the API. Then each secret engine and authentication method should have a list of methods (with links to javadoc) with short description, such as:

`TransitSecrets`
User methods:
- `Encrypt.Response encrypt(Encrypt.Request request)` - encrypt a message
- `Hmac.Response hmac(Hmac.Request request)` - create a HMAC of a message (similar to signatures, using symmetric cipher)
...
Configuration methods:
- `CreateKey.Response createKey(CreateKey.Request request)` - create RSA or AES key for encryption or signatures
...